docs/cli/app-types.md

```diff
@@ -0,0 +1,141 @@
+you will see that only your application's code and JSON.net have been published in the output. 
+The shared runtime remains outside of your application's dependency closure. 
+
+This type of application requires you to have a shared runtime installed. Otherwise, you will not be able 
+to run the application. 
+
+In order to create a portable application, all you need to do is to target the shared runtime in your `project.json`
+and have your frameworks aligned as the below sample shows. 
+
+```json
+"dependencies": {
+    "Microsoft.NETCore.App": {  
+        "version": "1.0.0-*",
+        "type": "platform"
+    }
+},
+"frameworks": {
+    "netcoreapp1.0": {}
+}
+```
+
+### Portable application with native dependencies
+A sub-group of the above, this type is a portable application that has native dependencies specified 
+somewhere in its dependency chain. This application is as portable as all of its native dependencies 
+are portable. You will be able to run the application on any platform that your native dependencies can 
+run on. Prime example of this is Kestrel, the ASP.NET Core cross-platform web server. It is built on top of 
+[libuv](https://github.com/libuv/libuv) which is its native dependency. 
+
+When you publish a portable application that has a native dependency, the published output will contain 
+all the same things as the portable application described in the previous section. For native dependencies
+the published output will contain a folder for each [RID](#what-are-rids) that the native dependency supports 
+(and that exists in its NuGet package). 
+
+As said, the portable application with a native dependency is the same as a portable application, just with a 
+dependency that has native components. The sample `project.json` fragment below is showing that:
+
+```json
+"dependencies": {
+    "Microsoft.NETCore.App": {  
+        "version": "1.0.0-*",
+        "type": "platform"
+    }
+    "Microsoft.AspNetCore.Server.Kestrel": "1.0.0-*"
+},
+"frameworks": {
+    "netcoreapp1.0": {}
+}
+```
+
+## Self-contained application
+Unlike the portable application, a *self-contained application* does not rely on any shared component to 
+be present on the machine where you want to deploy the application. As its name implies, it means that 
+the entire dependency closure, **including the runtime** is packaged with the application. This makes 
+it larger, but also makes it capable of running on any .NET Core supported platforms with the correct 
+native dependencies, regardless if it has the shared runtime installed or not. This makes it that much 
+easier to deploy to the target machine, since you only deploy your application. 
+
+Since the application carries the runtime with itself, you need to make a choice at the beginning of 
+development on which platforms you want to support. For instance, if you publish a self-contained application 
+for Windows 10, that same application will not work on OS X or Linux and vice versa. 
```

Comment:
Is this really a choice I am locked into? If I build my app and later decide to be stand-alone, why can't I add Ubuntu to my RIDs then?